### PR TITLE
Allow users with block perms to still use the `rescan_values` APIs if they have DB details or data model perms

### LINKED ISSUE #21764: User with `Manage database` permission can't run scan process
A user with Manage database permission and no data access to a database can't update field values.

The query fails due to lack of permissions.

Logs:
```
2022-04-19 14:25:09,253 ERROR models.field-values :: Error fetching field values
clojure.lang.ExceptionInfo: You do not have permissions to run this query. {:type :missing-required-permissions, :required-permissions #{"/db/1/schema/PUBLIC/table/1/query/"}, :actual-permissions #{"/data-model/db/1/" "/collection/3/" "/details/db/1/" "/block/db/1/" "/block/db/2/" "/data-model/db/2/"}, :permissions-error? true}
	at metabase.query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:30)
	at metabase.query_processor.middleware.permissions$perms_exception.doInvoke(permissions.clj:24)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:27)
	at metabase.query_processor.middleware.permissions$perms_exception.invoke(permissions.clj:24)
	at metabase.query_processor.middleware.permissions$fn__47056$check_ad_hoc_query_perms__47061$fn__47062.invoke(permissions.clj:84)
	at metabase.query_processor.middleware.permissions$fn__47056$check_ad_hoc_query_perms__47061.invoke(permissions.clj:80)
	at metabase.query_processor.middleware.permissions$fn__47087$check_query_permissions_STAR___47092$fn__47093.invoke(permissions.clj:99)
	at metabase.query_processor.middleware.permissions$fn__47087$check_query_permissions_STAR___47092.invoke(permissions.clj:89)
	at metabase.query_processor.middleware.permissions$check_query_permissions$fn__47106.invoke(permissions.clj:108)
	at metabase_enterprise.advanced_permissions.query_processor.middleware.permissions$check_download_permissions$fn__82728.invoke(permissions.clj:124)
	at metabase_enterprise.sandbox.query_processor.middleware.column_level_perms_check$maybe_apply_column_level_perms_check$fn__84500.invoke(column_level_perms_check.clj:30)
	at metabase.query_processor.middleware.mbql_to_native$mbql__GT_native$fn__50264.invoke(mbql_to_native.clj:23)
	at metabase.query_processor$fn__52796$combined_post_process__52801$combined_post_process_STAR___52802.invoke(query_processor.clj:207)
	at metabase.query_processor$fn__52796$combined_pre_process__52797$combined_pre_process_STAR___52798.invoke(query_processor.clj:204)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51244$fn__51249.invoke(resolve_database_and_driver.clj:35)
	at metabase.driver$do_with_driver.invokeStatic(driver.clj:75)
	at metabase.driver$do_with_driver.invoke(driver.clj:71)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51244.invoke(resolve_database_and_driver.clj:34)
	at metabase.query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47349.invoke(fetch_source_query.clj:281)
	at metabase.query_processor.middleware.store$initialize_store$fn__47537$fn__47538.invoke(store.clj:11)
	at metabase.query_processor.store$do_with_store.invokeStatic(store.clj:44)
	at metabase.query_processor.store$do_with_store.invoke(store.clj:38)
	at metabase.query_processor.middleware.store$initialize_store$fn__47537.invoke(store.clj:10)
	at metabase.query_processor.middleware.normalize_query$normalize$fn__51516.invoke(normalize_query.clj:22)
	at metabase_enterprise.audit_app.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__58093.invoke(handle_audit_queries.clj:130)
	at metabase.query_processor.reducible$async_qp$qp_STAR___44086$thunk__44088.invoke(reducible.clj:100)
	at metabase.query_processor.reducible$async_qp$qp_STAR___44086.invoke(reducible.clj:106)
	at metabase.query_processor.reducible$async_qp$qp_STAR___44086.invoke(reducible.clj:91)
	at metabase.query_processor.reducible$async_qp$qp_STAR___44086.invoke(reducible.clj:88)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.query_processor.reducible$sync_qp$qp_STAR___44097.doInvoke(reducible.clj:126)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.query_processor$process_query.invokeStatic(query_processor.clj:242)
	at metabase.query_processor$process_query.doInvoke(query_processor.clj:236)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at metabase.db.metadata_queries$qp_query$fn__66083.invoke(metadata_queries.clj:21)
	at metabase.db.metadata_queries$qp_query.invokeStatic(metadata_queries.clj:20)
	at metabase.db.metadata_queries$qp_query.invoke(metadata_queries.clj:18)
	at metabase.db.metadata_queries$field_query.invokeStatic(metadata_queries.clj:31)
	at metabase.db.metadata_queries$field_query.invoke(metadata_queries.clj:29)
	at metabase.db.metadata_queries$fn__66103$field_distinct_values__66112$fn__66115.invoke(metadata_queries.clj:76)
	at metabase.db.metadata_queries$fn__66103$field_distinct_values__66112.invoke(metadata_queries.clj:69)
	at metabase.db.metadata_queries$fn__66103$field_distinct_values__66112$fn__66113.invoke(metadata_queries.clj:73)
	at metabase.db.metadata_queries$fn__66103$field_distinct_values__66112.invoke(metadata_queries.clj:69)
	at clojure.lang.Var.invoke(Var.java:384)
	at metabase.models.field_values$distinct_values.invokeStatic(field_values.clj:132)
	at metabase.models.field_values$distinct_values.invoke(field_values.clj:124)
	at metabase.models.field_values$create_or_update_field_values_BANG_.invokeStatic(field_values.clj:154)
	at metabase.models.field_values$create_or_update_field_values_BANG_.doInvoke(field_values.clj:148)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at metabase.sync.field_values$fn__65980$update_field_values_for_field_BANG___65985$fn__65986.invoke(field_values.clj:23)
	at metabase.sync.field_values$fn__65980$update_field_values_for_field_BANG___65985.invoke(field_values.clj:21)
	at metabase.sync.field_values$fn__66006$update_field_values_for_table_BANG___66011$fn__66012$fn__66013$fn__66014.invoke(field_values.clj:44)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:158)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:151)
	at metabase.sync.field_values$fn__66006$update_field_values_for_table_BANG___66011$fn__66012$fn__66013.invoke(field_values.clj:42)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$reduce.invokeStatic(core.clj:6829)
	at clojure.core$reduce.invoke(core.clj:6812)
	at metabase.sync.field_values$fn__66006$update_field_values_for_table_BANG___66011$fn__66012.invoke(field_values.clj:41)
	at metabase.sync.field_values$fn__66006$update_field_values_for_table_BANG___66011.invoke(field_values.clj:38)
	at clojure.core$map$fn__5884.invoke(core.clj:2757)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.boundedLength(RT.java:1793)
	at clojure.lang.RestFn.applyTo(RestFn.java:130)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.field_values$fn__66033$update_field_values_for_database_BANG___66038$fn__66039.invoke(field_values.clj:52)
	at metabase.sync.field_values$fn__66033$update_field_values_for_database_BANG___66038.invoke(field_values.clj:50)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$fn__41434$run_step_with_metadata__41439$fn__41443$fn__41445.doInvoke(util.clj:388)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:100)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:94)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:117)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:114)
	at metabase.sync.util$fn__41434$run_step_with_metadata__41439$fn__41443.invoke(util.clj:383)
	at metabase.sync.util$fn__41434$run_step_with_metadata__41439.invoke(util.clj:378)
	at metabase.sync.util$fn__41650$run_sync_operation__41655$fn__41656$fn__41664.invoke(util.clj:489)
	at metabase.sync.util$fn__41650$run_sync_operation__41655$fn__41656.invoke(util.clj:487)
	at metabase.sync.util$fn__41650$run_sync_operation__41655.invoke(util.clj:481)
	at metabase.sync.field_values$fn__66059$update_field_values_BANG___66064$fn__66065$fn__66066.invoke(field_values.clj:67)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:158)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:151)
	at clojure.core$partial$fn__5859.invoke(core.clj:2634)
	at metabase.driver$fn__32587.invokeStatic(driver.clj:591)
	at metabase.driver$fn__32587.invoke(driver.clj:591)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__41343.invoke(util.clj:136)
	at metabase.sync.util$with_db_logging_disabled$fn__41340.invoke(util.clj:127)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:100)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:94)
	at metabase.sync.util$with_start_and_finish_logging$fn__41329.invoke(util.clj:112)
	at metabase.sync.util$with_sync_events$fn__41324.invoke(util.clj:86)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__41315.invoke(util.clj:65)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:179)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:176)
	at metabase.sync.field_values$fn__66059$update_field_values_BANG___66064$fn__66065.invoke(field_values.clj:65)
	at metabase.sync.field_values$fn__66059$update_field_values_BANG___66064.invoke(field_values.clj:61)
	at metabase.api.database$fn__75260$fn__75261.invoke(database.clj:756)
	at clojure.core$binding_conveyor_fn$fn__5772.invoke(core.clj:2034)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	```

### PR DESCRIPTION
**Pull request by noahmoss**
If a user has block perms for a data source, we prevent them from reading the `FieldValues` model since it contains actual source data.

However, if the user has DB details or data model permissions, we _should_ allow them to use the `rescan_values` APIs for a DB/table/field. This API doesn't return any data; it just triggers a sync on the backend.

I've fixed these APIs by simply clearing the bound `*current-user*` before initiating the sync.

Resolves #21764

### CODE REVIEW

File: src/metabase/server/middleware/session.clj [hunk 1]
```diff
@@ -189,8 +189,7 @@
                       [:user.is_superuser :is-superuser?]
                       [:user.locale :user-locale]]
           :from      [[Session :session]]
-          :left-join [[User :user] [:= :session.user_id :user.id]
-                      ]
+          :left-join [[User :user] [:= :session.user_id :user.id]]
```
Comment (noahmoss) [inline src/metabase/server/middleware/session.clj:192]: Unrelated formatting fix

**Review by qnkhuat — CHANGES_REQUESTED** [review]
Nice solution, it works for re-scan field values but still fails when I try to sync DB schema.

**Comment by luizarakaki** [comment]
@qnkhuat Did you try to sync the DB schema on the DB page? Eg: http://localhost:3000/admin/databases/1?

I could both rescan values and sync the db schema, no problems.

**Comment by qnkhuat** [comment]
hmmm, yea it works now...

**Review by qnkhuat — APPROVED** [review]

**Pull request merged**